gitlab-runner/src/lib.rs

```diff
@@ -76,11 +76,13 @@ pub mod uploader;
 pub use logging::GitlabLayer;
 use tokio_util::sync::CancellationToken;
 use tracing::instrument::WithSubscriber;
+pub use uploader::UploadFile;
 mod runlist;
 use crate::runlist::RunList;
-use uploader::Uploader;
 
 use futures::prelude::*;
+use futures::AsyncRead;
```

Comment:
As we depend on Tokia should we use futres::AsyncRead or Tokio::AsyncRead ; did you pick one over the others for a specific reason ooi

Reply:
Futures is meant to follow/proof out stuff that's going into stdlib as well indeed as ; But there were still quite some discussion around AsyncRead last i checked. 

I don't really have a strict preference here; there are also adapters both way around anyway; So i'm fine with going, use futures as external API where applicable (e.g. we use tokio for cancellation token as futures doesn't have that) :)